src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpRequestMessageHelper.cs

```diff
@@ -0,0 +1,367 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.ServiceModel.Diagnostics.Application;
+using System.ServiceModel.Security;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace System.ServiceModel.Channels
+{
+    internal class HttpResponseMessageHelper
```

Comment:
A lot of the code here came from HttpInput, with some conversion to async where necessary.